pkg/client/record/event.go

```diff
@@ -25,11 +26,23 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
 
 	"github.com/golang/glog"
 )
 
+const (
+	maxQueuedEvents  = 1000
+	maxTriesPerEvent = 10
+)
+
+var (
+	minSleep   = float64(1 * time.Second)
+	maxSleep   = float64(15 * time.Second)
+	backoffExp = 1.5
```

Comment:
The logic in the code you commented on is only governing the behavior while the master is still down. It would allow for a surge of requests when the master comes back up if the master isn't aggressive in returning errors, because this code only sleeps after a request fails. A success immediately resets the sleep to 0, which is potentially dangerous since each client is likely to have more than one event queued up.

I've implemented what you suggested, but it may be worth adding additional logic to slowly ease back onto the pedal once requests start succeeding again. Something along the lines of having an index into list of sleep times (or a counter + exponential backoff), and incrementing the index/counter foward or backward depending on whether each request succeeds.